hostsfile/src/lib.rs

```diff
@@ -265,10 +283,12 @@ impl HostsBuilder {
     }
 
     fn write_and_swap(temp_path: &Path, hosts_path: &Path, contents: &[u8]) -> io::Result<()> {
-        // Copy the file we plan on modifying so its permissions and metadata are preserved.
-        std::fs::copy(hosts_path, temp_path)?;
-        Self::write_clobber(temp_path, contents)?;
-        std::fs::rename(temp_path, hosts_path)?;
+        if Self::has_content_changed(hosts_path, contents)? {
```

Comment:
Just a thought but not necessarily something we should do in this PR:

Have these write functions return a `bool` indicating if we actually wrote the hosts file, and only output a message if we do.

Currently `innernet` unconditionally prints:

```
[*] updating /etc/hosts with the latest peers.
```

Even if there is nothing to change.

Reply:
That's a great idea!

So I took a look at `client/src/main.rs` and thought about moving the `updating /etc/hosts` log to an `else` statement [here](https://github.com/tonarino/innernet/blob/ae96e05e903a9946f9da4579baac11fef7ad9ecf/client/src/main.rs#L294), also the message should say `updated /etc/hosts` instead.

Do you think there should be a log at the beginning of `update_hosts_file` with something like `trying to update /etc/hosts` or should we use this moment to reduce the number of logs related to `/etc/hosts`?

Reply:
Just my opinion, but I think we should only output something related to `/etc/hosts` if we actually do something to it.

Right now we output `updating /etc/hosts with the latest peers`, but it's not like the user has any chance to stop it before it writes, so we might as well just output a message when we update it, or not output anything if we don't.

@mcginty @strohel what do you think?

Reply:
Fixed!